packages/cadl-ranch-specs/http/models/inheritance/main.cadl

```diff
@@ -1,60 +1,253 @@
-namespace BasicPolymorphicModels;
+@doc("Illustrates inheritance and polymorphic model.")
+@scenarioService("/models/inheritance")
+namespace Models.Inheritance;
 
-@doc("Example base type.")
-model BaseClass {
-  @doc("An example property.")
-  baseClassProperty: string;
+@doc("This is base model for non-discriminator normal multiple levels inheritance.")
+model Pet {
+  name: string;
 }
 
-model DerivedFromBaseClassA extends BaseClass {
-  @doc("An example property on a derived type")
-  derivedClassAProperty: string;
+@doc("The second level model in the normal multiple levels inheritance.")
+model Cat extends Pet {
+  age: int32;
 }
 
-model DerivedFromBaseClassB extends BaseClass {
-  @doc("An example property on a derived type")
-  derivedClassBProperty: string;
+@doc("The third level model in the normal multiple levels inheritance.")
+model Siamese extends Cat {
+  smart: boolean;
 }
 
-@discriminator("discriminatorProperty")
-@doc("Example base class that has a discriminator property.")
-model BaseClassWithDiscriminator extends BaseClass {}
-// Note: discriminator property is not defined on parent
+@doc("This is base model for polymorphic multiple levels inheritance with a discriminator.")
+@discriminator("kind")
+model Fish {
+  age: int32;
+}
+
+@doc("The second level model in polymorphic multiple levels inheritance and it defines a new discriminator.")
+@discriminator("sharktype")
+model Shark extends Fish {
+  kind: "shark";
+}
 
-model DerivedFromBaseClassWithDiscriminatorA extends BaseClassWithDiscriminator {
-  discriminatorProperty: "A";
+@doc("The second level model in polymorphic multiple levels inheritance which contains references to other polymorphic instances.")
+model Salmon extends Fish {
+  kind: "salmon";
+  friends?: Fish[];
+  hate?: Record<Fish>;
+  partner?: Fish;
 }
 
-model DerivedFromBaseClassWithDiscriminatorB extends BaseClassWithDiscriminator {
-  discriminatorProperty: "B";
+@doc("The third level model SawShark in polymorphic multiple levels inheritance.")
+model SawShark extends Shark {
+  sharktype: "saw";
 }
 
-@doc("Illustrates case where a basic model has polymorphic properties.")
-model ModelWithPolymorphicProperty {
-  @doc("Example polymorphic type property.")
-  polymorphicProperty: BaseClassWithDiscriminator;
+@doc("The third level model GoblinShark in polymorphic multiple levels inheritance.")
+model GoblinShark extends Shark {
+  sharktype: "goblin";
 }
 
 @scenario
-@scenarioDoc("Generate, send, and receive round-trip inherited model.")
-@route("/model")
+@scenarioDoc("""
+Generate and send model.
+Expected input body:
+```json
+{"name": "abc", "age": 32, "smart": true}
+```
+""")
+@route("/valid")
+@post
+op postValid(@body input: Siamese): OkResponse | ErrorResponse;
+
+@scenario
+@scenarioDoc("""
+Generate and receive model.
+Expected response body:
+```json
+{"name": "abc", "age": 32, "smart": true}
+```
+""")
+@route("/valid")
+@get
+op getValid(): Siamese | ErrorResponse;
```

Comment:
I would remove `| ErrorResponse` from all the operations. We should not care about errors in most specs.( if there is an error, it is unexpected and the test fail)

Reply:
removed all `| ErrorResponse` since there is no error tests